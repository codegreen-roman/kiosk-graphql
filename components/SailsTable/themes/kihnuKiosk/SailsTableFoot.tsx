import React, { FC } from 'react'
import { useTranslation } from 'react-i18next'
import { Icon } from '@components/Icon'
import { Root, Message, IconHolder } from './SailsTableFootStyles'

export interface SailsTableFootProps {
  hasDangerGoods: boolean
  hasLocalPassengers: boolean
  hasCanceledSails: boolean
  isTicketsSoldOut: boolean
  hasRestrictedPrice: boolean
  hasReserveTickets: boolean
  isSomeReserveTicketsSoldOut: boolean
  noLocalsWarningFromAttributes: string
  warnings: string[]
}

const SailsTableHead: FC<SailsTableFootProps> = ({
  isTicketsSoldOut,
  hasLocalPassengers,
  hasDangerGoods,
  hasCanceledSails,
  hasRestrictedPrice,
  hasReserveTickets,
  isSomeReserveTicketsSoldOut,
  noLocalsWarningFromAttributes,
  warnings,
}) => {
  const { t } = useTranslation()

  return (
    <Root>
      <Message data-test="sold-out" display-if={isTicketsSoldOut} type="warning">
        {warnings && warnings.map((warning, index) => warning && (index === 0 ? `* ${warning} \n` : warning))}
      </Message>

      <Message data-test="local" display-if={hasLocalPassengers} type="annotation">
        {t('Presale')}
      </Message>

      <Message data-test="danger" display-if={hasDangerGoods} type="danger">
        <IconHolder as={Icon.danger} />
        {t('Dangerous goods')}
      </Message>

      <Message data-test="cancel" display-if={hasCanceledSails} type="error">
        <IconHolder as={Icon.anchor} />
        {t('Sail was cancelled')}
      </Message>

      <Message data-test="cancel" display-if={hasRestrictedPrice} type="default">
        <IconHolder as={Icon.noTrucks} />
        {t('Heavy trucks')}
      </Message>

      <Message data-test="locals-description" display-if={hasReserveTickets} type="default">
        {t('For locals')}
      </Message>

      <Message
        data-test="no-locals"
        display-if={isSomeReserveTicketsSoldOut && noLocalsWarningFromAttributes}
        type="danger"
      >
        ** {noLocalsWarningFromAttributes}
      </Message>
    </Root>
  )
}

export default SailsTableHead
