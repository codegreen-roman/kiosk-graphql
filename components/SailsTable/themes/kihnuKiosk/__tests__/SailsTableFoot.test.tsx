import React, { ReactNode } from 'react'
import SailsTableFoot, { SailsTableFootProps } from '../SailsTableFoot'
import { shallow } from 'enzyme'

jest.mock('react-i18next', () => ({
  useTranslation: (): object => ({
    t: (): string => 'Translation',
    i18n: {
      language: 'et',
    },
  }),
}))

jest.mock('../../../../../hooks/useAppState', () => ({
  useAppState: (): object => ({
    state: {
      route: {
        code: 'SVI-ROH-SVI',
        attributes: [
          {
            code: 'KIOSK_SOLD_OUT_INFO_PART_TWO_FI',
            value:
              'Lippu tavalliselle reitille voi ostaa kioskista ennen lähtöä {{CASH_DESK_HIGH_METERS_LIMIT}} lineaarimetrillä {{CASH_DESK_DECK_LIMIT}} matkustajalle. Liput lisätään, kun ajastin nollataan.',
            __typename: 'RouteAttribute',
          },
          {
            code: 'KIOSK_SOLD_OUT_INFO_PART_TWO_RU',
            value:
              'Билет на регулярный рейс можно купить в киоске перед отправлением из расчёта {{CASH_DESK_HIGH_METERS_LIMIT}} линейных метров на {{CASH_DESK_DECK_LIMIT}} пассажиров. Билеты будут добавлены, когда таймер обнулится',
            __typename: 'RouteAttribute',
          },
          {
            code: 'KIOSK_SOLD_OUT_INFO_PART_ONE_DE',
            value: 'Tickets für den Vorverkauf sind ausverkauft.',
            __typename: 'RouteAttribute',
          },
          {
            code: 'KIOSK_SOLD_OUT_INFO_PART_ONE_ET',
            value: 'EELMÜÜGI KOHAD ON VÄLJA MÜÜDUD.',
            __typename: 'RouteAttribute',
          },
          {
            code: 'KIOSK_SOLD_OUT_INFO_PART_ONE_EN',
            value: 'Changed text. PRE-BOOKABLE CAPACITY IS SOLD OUT.',
            __typename: 'RouteAttribute',
          },
          {
            code: 'KIOSK_SOLD_OUT_INFO_PART_TWO_ET',
            value:
              'REGULAARREISIDEL MÜÜAKSE KIOSKIST ENNE LAEVA VÄLJUMIST {{CASH_DESK_DECK_LIMIT}} REISIJA KOHTA JA {{CASH_DESK_HIGH_METERS_LIMIT}} LIINIMEETRIT TEKIPINDA. KOHAD LISATAKSE, KUI TAIMER AEGUB.',
            __typename: 'RouteAttribute',
          },
          {
            code: 'KIOSK_SOLD_OUT_INFO_PART_TWO_EN',
            value:
              'Changed text. WHEN THE TIMER EXPIRES, ON THE REGULAR TRIPS IT WILL BE ADDED {{CASH_DESK_DECK_LIMIT}} PASSENGER TICKETS AND {{CASH_DESK_HIGH_METERS_LIMIT}} LINE METERS OF THE DECK BEFORE THE DEPARTURE.',
            __typename: 'RouteAttribute',
          },
          {
            code: 'KIOSK_SOLD_OUT_INFO_PART_TWO_DE',
            value:
              'Tickets für reguläre Fahrten können vor der Abfahrt im Kiosk in Berechnung von {{CASH_DESK_HIGH_METERS_LIMIT}} laufenden Metern für {{CASH_DESK_DECK_LIMIT}} Passagiere erworben werden. Die Tickets werden hinzugefügt, wenn der Timer auf Null geht.',
            __typename: 'RouteAttribute',
          },
          {
            code: 'KIOSK_SOLD_OUT_INFO_PART_ONE_FI',
            value: 'Ennakkoliput ovat loppuneet.',
            __typename: 'RouteAttribute',
          },
          {
            code: 'KIOSK_SOLD_OUT_INFO_PART_ONE_RU',
            value: 'Билеты для предварительной продажи закончились.',
            __typename: 'RouteAttribute',
          },
          {
            code: 'KIOSK_SOLD_OUT_INFO_PART_TWO_SV',
            value:
              'ALLA FÖRBOKBARA BILJETTER ÄR SLUTSÅLDA. NÄR TIMERN RÄKNAT NER, BLIR {{CASH_DESK_DECK_LIMIT}}  PASSAGERARBILJETTER OCH {{CASH_DESK_HIGH_METERS_LIMIT}} METER FORDONSPLATS YTTERLIGARE TILLGÄNGLIGA',
            __typename: 'RouteAttribute',
          },
          {
            code: 'KIOSK_SOLD_OUT_INFO_PART_ONE_SV',
            value: 'ALLA FÖRBOKBARA BILJETTER ÄR SLUTSÅLDA.',
            __typename: 'RouteAttribute',
          },
        ],
      },
    },
  }),
}))

describe('SailsTableHead', () => {
  const defaultProps: SailsTableFootProps = {
    hasDangerGoods: true,
    hasLocalPassengers: true,
    hasCanceledSails: true,
    isTicketsSoldOut: true,
    hasRestrictedPrice: false,
    hasReserveTickets: false,
    isSomeReserveTicketsSoldOut: false,
    noLocalsWarningFromAttributes: undefined,
    warnings: [
      'EELMÜÜGI KOHAD ON VÄLJA MÜÜDUD.',
      'REGULAARREISIDEL MÜÜAKSE KIOSKIST ENNE LAEVA VÄLJUMIST {{CASH_DESK_DECK_LIMIT}} REISIJA KOHTA JA {{CASH_DESK_HIGH_METERS_LIMIT}} LIINIMEETRIT TEKIPINDA. KOHAD LISATAKSE, KUI TAIMER AEGUB.',
    ],
  }
  let component

  function render(props?: Partial<SailsTableFootProps>): ReactNode {
    const updatedProps: SailsTableFootProps = {
      ...defaultProps,
      ...props,
    }
    return shallow(<SailsTableFoot {...updatedProps} />)
  }

  describe('when default props passed', () => {
    beforeAll(() => {
      component = render()
    })

    it('matches snapshot', () => {
      expect(component).toMatchSnapshot()
    })

    it('renders sold-out message', () => {
      expect(component.find('[data-test="sold-out"]')).toExist()
    })

    it('renders local message', () => {
      expect(component.find('[data-test="local"]')).toExist()
    })

    it('renders danger message', () => {
      expect(component.find('[data-test="danger"]')).toExist()
    })

    it('renders cancel message', () => {
      expect(component.find('[data-test="cancel"]')).toExist()
    })
  })

  describe('when hasDangerGoods prop is false', () => {
    beforeAll(() => {
      component = render({
        hasDangerGoods: false,
      })
    })

    it('does not render danger message', () => {
      expect(component.find('[data-test="danger"]')).not.toExist()
    })
  })

  describe('when isTicketsSoldOut prop is false', () => {
    beforeAll(() => {
      component = render({
        isTicketsSoldOut: false,
      })
    })

    it('does not render sold-out message', () => {
      expect(component.find('[data-test="sold-out"]')).not.toExist()
    })
  })

  describe('when hasLocalPassengers prop is false', () => {
    beforeAll(() => {
      component = render({
        hasLocalPassengers: false,
      })
    })

    it('does not render local message', () => {
      expect(component.find('[data-test="local"]')).not.toExist()
    })
  })

  describe('when hasCanceledSails prop is false', () => {
    beforeAll(() => {
      component = render({
        hasCanceledSails: false,
      })
    })
  })
})
