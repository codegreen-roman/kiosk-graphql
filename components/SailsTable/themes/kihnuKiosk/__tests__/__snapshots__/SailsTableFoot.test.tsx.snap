// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SailsTableHead when default props passed matches snapshot 1`] = `
<SailsTableFootStyles__Root>
  <SailsTableFootStyles__Message
    data-test="sold-out"
    type="warning"
  >
    * EELMÜÜGI KOHAD ON VÄLJA MÜÜDUD. 

    REGULAARREISIDEL MÜÜAKSE KIOSKIST ENNE LAEVA VÄLJUMIST {{CASH_DESK_DECK_LIMIT}} REISIJA KOHTA JA {{CASH_DESK_HIGH_METERS_LIMIT}} LIINIMEETRIT TEKIPINDA. KOHAD LISATAKSE, KUI TAIMER AEGUB.
  </SailsTableFootStyles__Message>
  <SailsTableFootStyles__Message
    data-test="local"
    type="annotation"
  >
    Translation
  </SailsTableFootStyles__Message>
  <SailsTableFootStyles__Message
    data-test="danger"
    type="danger"
  >
    <SailsTableFootStyles__IconHolder
      as={[Function]}
    />
    Translation
  </SailsTableFootStyles__Message>
  <SailsTableFootStyles__Message
    data-test="cancel"
    type="error"
  >
    <SailsTableFootStyles__IconHolder
      as={[Function]}
    />
    Translation
  </SailsTableFootStyles__Message>
</SailsTableFootStyles__Root>
`;
